woocommerce/woocommerce
# In-app Marketplace React rebuild: feature branch

### Changes proposed in this Pull Request:





This feature branch contains a collection of work that leads to replacing the existing PHP-powered In-app Marketplace (the one you see when visiting WooCommerce > Extensions) with one powered by React and integrated into WC Admin.

The changes include a new design based on the current design language of WooCommerce admin screens.

This is the first phase that seeks to mostly replicate existing functionality and make minimal improvements to release an MVP. This will act as a base to make further improvements to the In-app Marketplace experience.

Changes:

- Rebuilt the Extensions (Discover) page in React with new design
- Rebuilt the search (Browse) page
- Hooked up the existing PHP "My Subscriptions" page until a later phase rebuilds this in React too
- Adds a feature flag that can be toggled in WooCommerce > Settings > Advanced > Features to keep the old design



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Pull down a copy of this branch. 
2. Build WooCommerce on your local environment, e.g.: `cd plugins/woocommerce` then `nvm use && pnpm run build --filter=woocommerce`.
3. Refresh WP Admin and go to `WooCommerce > Extensions`
4. You should see the new In-app marketplace and be able to use it fully.

#### Discover section populated with products is the first you see
![Screen Shot 2023-08-17 at 14 51 51](https://github.com/woocommerce/woocommerce/assets/22053773/c18a5ff9-e920-4918-a7be-432aaf4d58db)

#### The "Browse" tab displays products with category filters
![Screen Shot 2023-08-17 at 14 54 30](https://github.com/woocommerce/woocommerce/assets/22053773/5c493901-219b-4725-ae64-1d3c66a6a253)

#### Searching should display relevant products
![Screen Shot 2023-08-17 at 14 56 26](https://github.com/woocommerce/woocommerce/assets/22053773/cd11d108-669b-4d2e-b869-693314253517)

#### My Subscriptions tab functions correctly
Clicking the "My Subscriptions" tab should redirect you to the existing PHP My Subscriptions page.

![Screen Shot 2023-08-17 at 14 57 55](https://github.com/woocommerce/woocommerce/assets/22053773/3da819d4-1736-4d4b-bdc5-d42cc622bd7b)

#### You can connect and disconnect a WooCommerce.com account
**Click the user menu (top-left next to the search) to connect a WooCommerce.com account**
![Screen Shot 2023-08-17 at 14 59 44](https://github.com/woocommerce/woocommerce/assets/22053773/3a3bf7a6-cc48-42b8-8b68-41fbc210fd85)

**Return to the In-app marketplace once connected and use the user menu to disconnect your WooCommerce.com account**
![Screen Shot 2023-08-17 at 14 59 18](https://github.com/woocommerce/woocommerce/assets/22053773/38122457-95c1-476c-ad03-17a7c5df0434)

![Screen Shot 2023-08-17 at 14 59 22](https://github.com/woocommerce/woocommerce/assets/22053773/8b3363fd-4452-481f-a17b-e60bd233ac79)

#### Disable the new feature
**Disable the feature from `WooCommerce > Settings > Advanced > Features`** to see the old In-app Marketplace:
<img width="1252" alt="Screenshot 2023-08-17 at 15 03 32" src="https://github.com/woocommerce/woocommerce/assets/22053773/5e0167b7-3df6-4942-9654-672286240785">

![Screen Shot 2023-08-17 at 15 04 31](https://github.com/woocommerce/woocommerce/assets/22053773/8223c0eb-5697-477a-890a-5e4e070a73b3)



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

We have completely redesigned the In-app Marketplace.

#### Comment 

</details>
